tools/stress-cluster/services/Stress.Watcher/src/PodEventHandler.cs

```diff
@@ -215,5 +228,78 @@ public bool ShouldStartPodChaos(WatchEventType type, V1Pod pod)
 
             return true;
         }
+
+        public async Task DeleteResources(V1Pod pod)
+        {
+            if (!ShouldDeleteResources(pod))
+            {
+                Logger.Debug($"Skipping pod deletion for pod {pod.Name()}.");
+                return;
+            }
+
+            var rgName = GetResourceGroupName(pod);
+
+            if (rgName == "")
+            {
+                return;
+            }
+
+            Subscription subscription = ARMClient.DefaultSubscription;
+
+            ResourceGroup resourceGroup;
+            try {
+                resourceGroup = await subscription.GetResourceGroups().GetAsync(rgName);
+            } catch (Exception e){
+                Logger.Error($"Failed to get resource group {rgName} using subsription id {subscription.Id}");
+                Logger.Error($"Throwing exception: {e}");
+                return;
+            }
+
+            await resourceGroup.DeleteAsync();
+            Logger.Information($"Deleting Resources {rgName}");
+        }
+
+        public bool ShouldDeleteResources(V1Pod pod)
+        {
+            if (!string.IsNullOrEmpty(Namespace) && Namespace != pod.Namespace())
+            {
+                return false;
+            }
+
+            var initContainers = pod.Spec?.InitContainers;
+            if (initContainers == null || initContainers.Count() == 0) {
+                return false;
+            }
+
+            var deployContainers = initContainers.Where(c => c.Name == "init-azure-deployer");
+            if (deployContainers.Count() == 0)
+            {
+                return false;
+            }
+
+            return (pod.Status.Phase == "Succeeded") || (pod.Status.Phase == "Failed");
+        }
+
+        public string GetResourceGroupName(V1Pod pod)
+        {
+            var deployContainers = pod.Spec.InitContainers.Where(c => c.Name == "init-azure-deployer");
+            var envVars = deployContainers.First().Env;
+            if (envVars == null) {
+                return "";
+            }
```

Comment:
To be thorough, I would also handle null `InitContainers` here, but you are protected right now by the fact that you already did a similar null check in `ShouldDeleteResources`

Something like

```suggestion
            var deployContainers = pod.Spec.InitContainers?.Where(c => c.Name == "init-azure-deployer");
            var envVars = deployContainers?.First().Env;
            if (envVars == null) {
                return "";
            }
```